pkg/hooks/v1alpha1/api_v1alpha1.proto

```diff
@@ -3,7 +3,9 @@ syntax = "proto3";
 package kubevirt.hooks.v1alpha1;
 
 service Callbacks {
-    rpc OnDefineDomain (OnDefineDomainParams) returns (OnDefineDomainResult);
+    rpc OnDefineDomain (OnDefineDomainParams) returns (OnDefineDomainResult) {
+        option deprecated = true;
+    };
```

Comment:
If I understand correctly, this will produce a build-time warning if someone uses the call, right? I think it would be nice to have something at runtime as well. Perhaps the VM admission webhook can generate a warning if `hooks.kubevirt.io/hookSidecars` annotation is used with `--version v1alpha1`.